### PR TITLE
Introduce SubscanV2 API

### PR DESCRIPTION
**Pull request by Tomen**
This PR introduces a SubscanV2 API Wrapper as a configurable option. In a future version of Subscrape, this should become the default version.

## What's new?
- You can now configure the API which shall be used for scraping by setting the `_api` param in the configuration within a chain.
- `SubscrapeDB` is rewritten to store hydrated extrinsics and events in a dedicated Sqlite file. Previous storage are being redefined as indexes (or summary elements)
- New operations `extrinsics-list` and `events-list` were added. You can submit lists of extrinsic or event indices and receive hydrated elements.
- `events` and `extrinsics` scraping: `module` can now be set to `None` to scrape all extrinsics or events. `name` can now be set to `None` to scrape all calls from the module.
- The improved paging mechanism of SubscanV2 is used
- the 'transfers' operation was fixed

This changes the output format of the results. Any code that uses the new scraper might break.

### Configuring the new `SubscanV2` API
```
config = {
        "kusama":{
            "_api": "SubscanV2",
            "extrinsics":{
                "crowdloan": ["create"]
            },
            "extrinsics-list":[
                "14238250-2"
            ],
            "events":{
                "crowdloan": ["created"]
            }
            "events-list":[
                "14238250-39"
            ],
            "transfers": [
                "Dp27W3mGpkT7BG9SxNsTtWoKcvJSwzF4BU8tadYirkn6Kwx"
            ]
        },
    }
```

For now, SubscanV1 is still the default version and is used automatically if `_api` is not given. Though at some point in the future, the default my change to SubscanV2 and at some point support for SubscanV1 might be discontinued.

## What changed
`SubscanWrapper` was refactored to be `SubscanBase`, from which `SubscanV1` and `SubscanV2` now inherit. Most logic is in `SubscanBase` for now. The specific classes only overwrite params atm but in the future I suspect specific method implementations will follow (paging changes in V2)

A bit of the logic from `ParachainScraper` had to move into `SubscanBase`, namely the fetch functions that parameterized the call. I wanted to avoid code duplication or ugly forks in code.

- [x] Updated Documentation + sample config
- [x] Updated Tests

### CODE REVIEW

File: tests/test_transfers.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    subscrape.wipe_storage()
+    logging.info("scraping")
+    await subscrape.scrape(config)
+    logging.info("transforming")
+
+    db = SubscrapeDB("kusama")
+    transfers_storage = db.storage_manager_for_transfers(address)
+    transfers = dict(transfers_storage.get_iter())
+
+    first_transfer = transfers["7849231-5"]
```
Comment (spazcoin) [inline tests/test_transfers.py:34]: maybe first assert that that extrinsic was even found.

**Review by spazcoin — APPROVED** [review]
looks good to me

**Comment by spazcoin** [comment]
I was going to merge, but it looks like you need to merge `main` first.

File: tests/test_transfers.py (hunk 1 above)
Comment (Tomen) [reply tests/test_transfers.py:34]: my thinking is that it throws an exception either way, so why bother? or do you mean to provide more specific guidance to the developer?

**Pull request merged**